### DISCUSSION TITLE
[iOS17] Thermal state critical

### DISCUSSION THREAD

**Discussion by mkrn** [HaishinKit/HaishinKit.swift#1305] [Q&A]
When streaming on 5g getting thermal state Critical very quickly. What can be done to delay it from happening?
Seems like on WiFi it takes a bit longer to heat up.. 
Can we stop showing camera view? Or stop streaming but keep recording or vice versa?

**Comment by shogo4405** [comment]
Upon upgrading to iOS 17 or Xcode 15, I am also experiencing overheating on my end.

Please try:
Stop camera view.
- view.attachStream(nil)

Stop streaming
- stream.publish(nil)

**Comment by mkrn** [comment]
I experience overheating especially quickly on iPhone 13 Pro. every time on 4G/5G within 5-10 minutes, not direct sun (room temperature about 20-23 degrees) 
a little longer on wifi until overheat

With iphone 11 pro i can go on for a long time.

Is this related to new audio handling code ?

**Comment by shogo4405** [comment]
In my environment (iPhone 15 Pro Max + iOS 17), I am also experiencing overheating with version 1.6.0. I believe it's not related to changes in audio processing but rather due to iOS 17.

I believe you might not be using the multi-camera feature (https://github.com/shogo4405/HaishinKit.swift#multi-camera). In that case, by making this configuration, you can reduce CPU usage and mitigate overheating:

https://github.com/shogo4405/HaishinKit.swift/compare/main...feature/420YpCbCr#diff-29298f6137b27726fe4977b6fee532563f5bba1cfb69f4d911f7636bcbf36c00L7-L13

**Comment by shogo4405** [comment]
It seems there was an issue up to iOS 17.0-17.0.2. I updated to 17.0.3 on my end, and I observed improvements.
https://support.apple.com/en-us/HT213961

**Comment by yogeshdalavi** [comment]
i am still experiencing overheating of my iphone 13 with os26.3.1 and i am using haishinkit version2.2.5 please suggest me the solution in this.
